src/FixedRankMatrices.jl

```diff
@@ -78,14 +80,24 @@ together with its base point, see for example [`FixedRankMatrices`](@ref)
 * `UMVTVector(U,S,Vt,k)` store the umv factors after shortening them down to
   inner dimensions $k$, i.e. in $UMV^\mathrm{T}$, $M\in\mathbb R^{k\times k}$
 """
-struct UMVTVector{T} <: TVector
-    U::Matrix{T}
-    M::Matrix{T}
-    Vt::Matrix{T}
+struct UMVTVector{TU<:AbstractMatrix, TM<:AbstractMatrix, TVt<:AbstractMatrix} <: TVector
+    U::TU
+    M::TM
+    Vt::TVt
 end
 UMVTVector(U,M,Vt,k::Int) = UMVTVector(U[:,1:k],M[1:k,1:k],Vt[1:k,:])
 UMVTVector(U,M,Vt) = UMVTVector{eltype(U)}(U,M,Vt)
 
+*(v::UMVTVector, s::N) where {N <: Number} = UMVTVector(v.U*s, v.M*s,  v.Vpt*s)
+*(s::N, v::UMVTVector) where {N <: Number} = UMVTVector(s*v.U, s*v.M, s*v.Vpt) 
+/(v::UMVTVector, s::N) where {N <: Number} = UMVTVector(v.U/s, v.M/s, v.Vt/s)
+/(s::N, v::UMVTVector) where {N <: Number} = UMVTVector(v.U/s, v.M/s, v.Vt/s)
```

Comment:
I think this should be the definition of the `\` method, right?

Reply:
I don't want to solve linear equations, hm, what do I want to do? I don't know, maybe we leave that line out until I know again.

Reply:
Since you don't need the `N` type parameter, you could simply write `s::Number`. And, yes, I think the order of arguments should be preserved:
```julia
*(v::UMVTVector, s::Number) = UMVTVector(v.U*s, v.M*s,  v.Vpt*s)
*(s::Number, v::UMVTVector) = UMVTVector(s*v.U, s*v.M, s*v.Vpt) 
/(v::UMVTVector, s::Number) = UMVTVector(v.U/s, v.M/s, v.Vt/s)
/(s::Number, v::UMVTVector) = UMVTVector(s/v.U, s/v.M, s/v.Vt)
```
just in case somebody comes along with noncommutative number types.

Reply:
`/(s::Number, v::AbstractVector)` actually calculates a pseudoinverse of `v` and multiplies it by `s` so I don't think it's what you want to do here. https://github.com/JuliaLang/julia/blob/6d26f14ede64cd949e715a76816df989b8bdedfc/stdlib/LinearAlgebra/src/generic.jl#L1110
`\(s::Number, v::AbstractVector)` divides `v` by `s`.

Reply:
Then I'll better switch to `\`, you're right.